cpp/src/interop/to_arrow.cu

```diff
@@ -378,13 +378,15 @@ std::shared_ptr<arrow::Array> dispatch_to_arrow::operator()<cudf::list_view>(
   auto children_meta =
     metadata.children_meta.empty() ? std::vector<column_metadata>{{}, {}} : metadata.children_meta;
   auto child_arrays = fetch_child_array(input_view, children_meta, ar_mr, stream);
-  if (child_arrays.empty()) {
+  if (child_arrays.empty() || child_arrays[0]->data()->length == 0) {
     // Empty list will have only one value in offset of 4 bytes
     auto tmp_offset_buffer = allocate_arrow_buffer(sizeof(int32_t), ar_mr);
     memset(tmp_offset_buffer->mutable_data(), 0, sizeof(int32_t));
 
+    std::shared_ptr<arrow::Array> data =
+      child_arrays.empty() ? std::make_shared<arrow::NullArray>(0) : child_arrays[1];
```

Comment:
Shouldn't the type of the empty child array be preserved here i.e. `arrow::MakeEmptyArray` or does that not matter?

Reply:
aiui if the child_arrays are empty, then the list column had no children, so there's no information from libcudf on what the element type is.  But perhaps @davidwendt to confirm.

Reply:
Technically an empty list column would normally have a single empty child
https://github.com/rapidsai/cudf/blob/04330f2e9e73ac71a86666c55d0fe7248eaf8db6/cpp/src/lists/lists_column_factories.cu#L86-L94
But it may also not have any child too. 
I don't know if Arrow specifically cares but would bias towards what the callers of this API need.

Reply:
> Shouldn't the type of the empty child array be preserved here i.e. arrow::MakeEmptyArray or does that not matter?

To follow up, we _are_ preserving the child array if `child_arrays` is not `empty()` (we use `child_arrays[1]` which will have the appropriate type. If `child_arrays` _is_ `empty()` then there is no information available.

Reply:
But, I went for best of both worlds by using `MakeEmptyArray` rather than hand-constructing the offset buffer.